### PR TITLE
MTL-1664, MTL-1697, MTL-1698 add ntp script to develop brach

### PR DESCRIPTION
**Pull request by jacobsalmela**
#### Summary and Scope

- Relates to MTL-1664, MTL-1697, MTL-1968

##### Issue Type

- Bugfix Pull Request
- RFE Pull Request

Adds the NTP script to the develop branch

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on metal (e.g. an internal system, with hardware) (x) (if yes, please include results or a description of the test)
- [ ] I tested this on vshasta (if yes, please include results or a description of the test)
 
#### Idempotency
 
Deployed the template several times.
 
#### Risks and Mitigations
 
Low.

### CODE REVIEW

File: boxes/ncn-common/files/scripts/common/chrony/csm_ntp.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    USER_DATA_FILE = '/var/lib/cloud/instance/user-data.txt'
+    DEFAULT_CONF = '/etc/chrony.conf'
+    CONF_DIR = '/etc/chrony.d'
+
+    token = get_token()
+    xname = get_xname()
+    bss_data = get_bss_data(token, xname)
+    # get all the ntp keys needed to render the template
+    try:
+        allow = bss_data["ntp"]["allow"]
```
Comment (rustydb) [inline boxes/ncn-common/files/scripts/common/chrony/csm_ntp.py:185]: Same as https://github.com/Cray-HPE/node-image-build/pull/249/files#r857640133

**Review by rustydb — CHANGES_REQUESTED** [review]

**Review by manderson-hpe — APPROVED** [review]
Approved with the changes that were already requested by @rustydb

File: boxes/ncn-common/files/scripts/common/chrony/csm_ntp.py (hunk 1 above)
Comment (jacobsalmela) [reply boxes/ncn-common/files/scripts/common/chrony/csm_ntp.py:185]: Fixed

**Comment by jacobsalmela** [comment]
Feedback implemented.  It was requested not to have this for 1.2, so only this PR will go.

**Comment by jacobsalmela** [comment]
Rebasing onto Craig's branch in #231

**Review by rustydb — APPROVED** [review]

**Pull request merged**